### PR TITLE
Add compat flag / bug check for games on old Adreno/GL affected by #16015

### PR DESCRIPTION
**Pull request by hrydgard**
See #16015

Partly derived from https://github.com/hrydgard/ppsspp/compare/master...unknownbrackets:ppsspp:adreno-deadlock

But this one surgically activates it only for the known affected games, to avoid any additional performance regressions in the release. Ugly, but not sure what else to do given the nebulous nature of the driver bug.

Entirely untested.

### CODE REVIEW

File: GPU/GLES/GPU_GLES.cpp [hunk 1]
```diff
@@ -186,6 +186,12 @@ u32 GPU_GLES::CheckGPUFeatures() const {
 		features |= GPU_USE_SINGLE_PASS_STEREO;
 	}
 
+	if (draw_->GetBugs().Has(Draw::Bugs::ADRENO_RESOURCE_DEADLOCK)) {
+		if (PSP_CoreParameter().compat.flags().OldAdrenoPixelDepthRoundingGL) {
+			features |= GPU_ROUND_FRAGMENT_DEPTH_TO_16BIT;
```
Comment (unknownbrackets) [inline GPU/GLES/GPU_GLES.cpp:191]: `CheckGPUFeaturesLate()` might add another flag (like `GPU_ROUND_DEPTH_TO_16BIT` or `GPU_SCALE_DEPTH_FROM_24BIT_TO_16BIT`), not sure if this is really a problem or not, though.  In theory those flags can coexist...

**Review by ghost — COMMENTED** [review]
1. Forgot to add this games https://github.com/hrydgard/ppsspp/issues/16015#issuecomment-1256343383 in compat.ini

File: GPU/GLES/GPU_GLES.cpp (hunk 1 above)
Comment (hrydgard) [reply GPU/GLES/GPU_GLES.cpp:191]: Right, I'll re-order it to do it after CheckGPUFeatureLate, and also check for vendor workarounds.

**Comment by hrydgard** [comment]
@Gamemulatorer Done!

**Comment by Gamemulatorer** [comment]
Might be better to close this following issue:
***Naruto Shippuden Ultimate Ninja Impact***
https://github.com/hrydgard/ppsspp/issues/12847
***Kingdom Hearts Birth By Sleep***
https://github.com/hrydgard/ppsspp/issues/11029
***Monster Hunter 3rd HD Portable***
https://github.com/hrydgard/ppsspp/issues/11410
***Persona 3 Portable***
https://github.com/hrydgard/ppsspp/issues/11355 and only focus on the remaining performance issue in #16015?

**Comment by hrydgard** [comment]
Yeah, that's a good idea.

**Comment by Gamemulatorer** [comment]
This seems doesnt work correctly in adreno 306 I got freeze in Naruto Shippuden, Phantasy Star Portable 1 and 2 using my vivo y51a snapdragon 410.

**Pull request merged**